src/backend/metal/src/native.rs

```diff
@@ -34,6 +34,7 @@ pub struct RenderPass {
     pub(crate) desc: metal::RenderPassDescriptor,
     pub(crate) attachments: Vec<pass::Attachment>,
     pub(crate) color_channels: Vec<Option<Channel>>,
+    pub(crate) depth_stencil_aspects: hal::format::Aspects,
```

Comment:
What's the reason we need to track this separately outside of `desc`? As an optimization for state switches later?

Reply:
In `create_framebuffer` we need to know which attachments to assign the depth/stencil texture. Would you think that checking for `descriptor.xxx_attachment().is_some()` would achieve the same thing cleaner?

Reply:
IMO less state tracking is preferable if there's no significant disadvantage